clc/store/store.go

```diff
@@ -0,0 +1,189 @@
+package store
+
+import (
+	"errors"
+	"sync"
+
+	badger "github.com/dgraph-io/badger/v4"
+
+	"github.com/hazelcast/hazelcast-commandline-client/internal/log"
+)
+
+var (
+	ErrKeyNotFound = errors.New("key not found")
+)
+
+type StoreAccessor struct {
+	store *Store
+	mu    sync.Mutex
+}
+
+func NewStoreAccessor(dir string, logger log.Logger) *StoreAccessor {
+	return &StoreAccessor{
+		store: newStore(dir, logger),
+	}
+}
+
+func (sa *StoreAccessor) WithLock(fn func(s *Store) (any, error)) (any, error) {
+	sa.mu.Lock()
+	defer sa.mu.Unlock()
+	if err := sa.store.open(); err != nil {
+		return nil, err
+	}
+	defer sa.store.close()
+	return fn(sa.store)
+}
+
+type Store struct {
+	dir    string
+	db     *badger.DB
+	logger log.Logger
+}
+
+func newStore(dir string, logger log.Logger) *Store {
+	return &Store{
+		dir:    dir,
+		logger: logger,
+	}
+}
+
+func (s *Store) open() error {
+	opts := badger.DefaultOptions(s.dir)
+	opts.Logger = badgerLogger{s.logger}
+	db, err := badger.Open(opts)
+	if err != nil {
+		return err
+	}
+	s.db = db
+	return nil
+}
+
+type badgerLogger struct {
+	logger log.Logger
+}
+
+func (l badgerLogger) Errorf(log string, args ...any) {
+	l.logger.Debugf(log, args...)
+}
+func (l badgerLogger) Warningf(log string, args ...any) {
+	l.logger.Debugf(log, args...)
+}
+func (l badgerLogger) Infof(log string, args ...any) {
+	l.logger.Debugf(log, args...)
```

Comment:
Does it make sense to add a prefix, like `Store ERROR: `, etc. to these log lines so it's possible to distinguish between them?